hub/api/tensorview.py

```diff
@@ -84,9 +84,68 @@ def __setitem__(self, slice_, value):
                 new_nums.extend([None] * (len(slice_) - len(new_nums)))
                 new_offsets.extend([0] * (len(slice_) - len(new_offsets)))
             for i in range(len(slice_)):
-                slice_[i] = combine(slice_[i], new_nums[i], new_offsets[i])
+                slice_[i] = self._combine(slice_[i], new_nums[i], new_offsets[i])
             if len(new_nums) > len(slice_):
-                slice_ += self.new_nums[len(slice_) - len(new_nums):]
+                slice_ += self.new_nums[len(slice_) - len(new_nums) :]
         else:
-            raise TypeError("type {} isn't supported in dataset slicing".format(type(slice_)))
+            raise TypeError(
+                "type {} isn't supported in dataset slicing".format(type(slice_))
+            )
         self.dataset._tensors[self.subpath][slice_] = value
+
+    def _combine(self, slice_, num=None, ofs=0):
```

Comment:
Too long function without proper documentation that explains, please split the function into smaller pieces and write documentation for each piece and overall the logic of the computation